datastore/elastic_search/query.py

```diff
@@ -47,6 +57,216 @@ def dictionary_query(connection, index_name, doc_type, entity_name, **kwargs):
+                ]
+            }
+        },
+        "aggs": {
+            "unique_values": {
+                "terms": {
+                    "field": "language_script.keyword",
+                    "size": 100000
+                }
+            }
+        },
+        "size": 0
+    }
+    kwargs = dict(
+        kwargs, body=data, doc_type=doc_type, size=ELASTICSEARCH_SEARCH_SIZE,
+        index=index_name, filter_path=['aggregations.unique_values.buckets.key']
+    )
+    search_results = _run_es_search(connection, **kwargs)
+    language_list = [bucket['key'] for bucket in search_results['aggregations']['unique_values']['buckets']]
+    return language_list
+
+
+def get_dictionary_records(connection, index_name, doc_type, entity_name, word_list=None, **kwargs):
+    """
+    """
+    data = {
+        "query": {
+            "bool": {
+                'must': [{
+                    "match": {
+                        "entity_data": entity_name
+                    }
+                }]
+            }
+        },
+        "size": 10000
+    }
+
+    query_list = []
+    if word_list is not None:
+        word_list_chunks = [word_list[i:i + 500] for i in xrange(0, len(word_list), 500)]
+        for chunk in word_list_chunks:
+            updated_query = copy.deepcopy(data)
+            updated_query['query']['bool']['must'].append({
+                "terms": {
+                    "value.keyword": chunk
+                }
+            })
+            query_list.append(updated_query)
+    else:
+        query_list.append(data)
+
+    results = []
+    for query in query_list:
+        search_kwargs = dict(kwargs, body=query, doc_type=doc_type,
+                             size=ELASTICSEARCH_SEARCH_SIZE, index=index_name, scroll='1m')
+        search_results = _run_es_search(connection, **search_kwargs)
+
+        # Parse hits
+        if search_results:
+            results.extend(search_results['hits']['hits'])
+
+    return results
+
+
+def delete_dictionary_records_by_word(connection, index_name, doc_type, entity_name, word_list=None, **kwargs):
+    """
+    """
+    results = get_dictionary_records(
+        connection=connection,
+        index_name=index_name,
+        doc_type=doc_type,
+        entity_name=entity_name,
+        word_list=word_list,
+        **kwargs
+    )
+
+    delete_bulk_queries = []
+    str_query = []
+    for record in results:
+        delete_dict = {
+            '_index': index_name,
+            '_type': doc_type,
+            '_id': record["_id"],
+            '_op_type': 'delete',
+        }
+        str_query.append(delete_dict)
+        if len(str_query) > ELASTICSEARCH_BULK_HELPER_MESSAGE_SIZE:
+            delete_bulk_queries.append(str_query)
+            str_query = []
+
+    if str_query:
+        delete_bulk_queries.append(str_query)
+
+    for delete_query in delete_bulk_queries:
+        helpers.bulk(connection, delete_query, stats_only=True, **kwargs)
+
+
+def add_data_elastic_search(
```

Comment:
shouldn't this be present in `datastore/elastic_search/populate.py`

Reply:
Done